reaper-adm-extension/src/reaper_adm/pluginmain.cpp

```diff
@@ -173,8 +173,8 @@ extern "C" {
         reaperFileMenu->insert(std::move(admFileMenu), std::make_shared<BeforeNamedItem>("Project templates"));
 
         // Warning if the hard-coded fallback menu positions don't match anymore
-        if (!reaperMainMenu->checkHardcodedPosition("File")) api->ShowMessageBox("Positions of File menu does not match hard-coded fallback index.", "ADM Extension Error", MB_ICONEXCLAMATION);
-        if (!reaperFileMenu->checkHardcodedPosition("Project templates")) api->ShowMessageBox("Positions of Project templates menu does not match hard-coded fallback index.", "ADM Extension Error", MB_ICONEXCLAMATION);
+        if (!reaperMainMenu->checkHardcodedPosition("File")) api->ShowMessageBox("Position of File menu does not match hard-coded fallback index.", "ADM Extension Error", MB_ICONEXCLAMATION);
```

Comment:
I don't think this will work on Windows. I assume this code runs whilst the REAPER splash screen is displayed, and REAPER gets stuck on the splash screen if `api->ShowMessageBox` is used whilst the splash screen is showing. This is because `ShowMessageBox` is blocking so must be dismissed before REAPER will continue. The problem is that something about the splash screen prevents the message box being displayed. Only seems to affect Windows.
There's an alternative function called `winhelpers::NonBlockingMessageBox` which can be used in these cases - an example of it's usage is here: https://github.com/ebu/ear-production-suite/blob/6f64314e11b02ff39102c8118554d82a4b8ce04b/reaper-adm-extension/src/reaper_adm/pluginmain.cpp#L63-L68

Reply:
Actually, I've just noticed its `MessageBox` I'm using there rather than `api->ShowMessageBox`. Maybe `api->ShowMessageBox` just works?

Reply:
`api->ShowMessageBox` blocks the splash screen, but in my tests (on windows) reaper always continued starting as soon as I closed the warning. As you mentioned `winhelpers::NonBlockingMessageBox` is a bit nicer there, however all these defines would clutter up the code a bit. But I can try to throw the warning and use `winhelpers::NonBlockingMessageBox`  at a central spot to avoid this.